File: Projeto Web/src/app/component/doacao-produto/interface/doacaoProduto.ts
import { Doacao } from "@app/component/doar/interface/doacaoInterface";
import { Produto } from "@app/component/produto/interface/produtoInterface";

export interface DoacaoProduto {
    id: number;
    doacaoId: number,
    produtoId: number;
    quantidade: number;
  }

File: Projeto back-end/src/main/java/com/atividadeextensionista/entidades/DoacaoProduto.java
package com.atividadeextensionista.entidades;

import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Entity
@Data
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class DoacaoProduto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "doacao_id")
    private Doacao doacao;

    @ManyToOne
    @JoinColumn(name = "produto_id")
    private Produto produto;

    private Integer quantidade;
}
